static/js/enketo/widgets/db-object-widget.js

```diff
@@ -102,9 +104,11 @@ define( function( require, exports, module ) {
             }
             if (_.isObject(value)) {
                 // recursively set fields for children
-                return updateFields(value, path, originatingKeyPath);
+                return updateFields(value, path, index, originatingKeyPath);
             }
-            var node = Enketo.getCurrentForm().model.node(path);
+
+            var node = Enketo.getCurrentForm().model.node('(' + path + ')[' + index + ']');
```

Comment:
This looks like it's more precedence than you need, but it's not! `[]` has higher precedence than paths, and so if you don't do this weird stuff happens that I don't logically understand, except as much as to understand that it's wrong.

Reply:
Did you know, the [node](https://github.com/enketo/enketo-core/blob/master/src/js/Form-model.js#L250) function takes an `index` parameter? I don't know if it'll do what you need but it's worth a try!

Reply:
Nice tip! It is also zero-indexed like God intended.